pkg/updater/updater.go

```diff
@@ -147,6 +152,12 @@ func (u *Updater) Start(ctx context.Context) <-chan struct{} {
 							return fmt.Errorf("failed to get commitment: %w", err)
 						}
 
+						l2Block, err := u.l2Client.BlockByNumber(ctx, new(big.Int).SetUint64(commitment.BlockNumber))
```

Comment:
That's a good point, we want to use the l2.blocknumber to get the timestamp.

The preconf timestamp is always going to be equal to the block timestamp unless we update the consensus rule. 

@chrmatt did recommended a potential solution in which, we use the miners local time and allow a slack of latency between when the miner/POA geth node receives the txn for block production, and has it check against the internal clock. If the txn has too significant a drift from it's internal clock, we reject the txn. 

This will ensure an equilibrium where everyone cheats, very similar to [timing games](https://ethresear.ch/t/timing-games-implications-and-possible-mitigations/17612) that occur in the current consensus  layer of Ethereum. This also gives MEV potential to the POA geth/validator that's selected during the round robin process. Though the extractable value across a 200ms block seems limited, if we continue this approach after moving to longer block times, it could become a bigger issue for us.

Reply:
Fixed the bug btw, I needed to use `commitment.BlockCommitedAt` as per the contract changes [here.](https://github.com/primevprotocol/contracts/pull/85/files#diff-1baf8b4ed78bc44d258027e01832954a004d424cd622bfe2446a62a9d371b6b4L73-L74)